flytekit/core/base_task.py

```diff
@@ -789,6 +810,13 @@ def disable_deck(self) -> bool:
         """
         return self._disable_deck
 
+    @property
+    def decks(self) -> List[DeckField]:
```

Comment:
I think it's better to call this `deck_fields`

```suggestion
    def decks_fields(self) -> List[DeckField]:
```

A user can add custom decks that are not a preconfigured deck in `DeckField`.